src/PatchOrchestrationApplication/NodeAgentNTService/src/Manager/TimerManager.cs

```diff
@@ -310,16 +327,21 @@ private bool CheckApplicationExists()
             return applicationExists;
         }
 
-        private void CreateNewCheckpointFile()
+        private void CreateNewCheckpointFile(bool updateAttempted = true)
         {
             CheckpointFileData checkpointFileData = new CheckpointFileData();
             checkpointFileData.schedulingDateTime = this.GetNextSchedulingTime();
+            if(updateAttempted)
+            {
+                checkpointFileData.lastAttemptedUpdateTime =  DateTime.UtcNow;
+            }
             checkpointFileData.rescheduleCount = 0;
             checkpointFileData.rescheduleNeeded = false;
             this.WriteCheckpointFile(checkpointFileData);
+            this.PostWUUpdateEventOnServicePackage();
         }
 
-        private bool UpdateSettingsAndCreateCheckpoint()
+        private bool UpdateSettingsAndCreateCheckpoint(bool updateAttempted = true)
```

Comment:
>UpdateSettingsAndCreateCheckpoint [](start = 21, length = 33)

Shouldnt you be preserving the last attempt in this case?

Reply:
I am doing it